File: app/src/main/java/me/wcy/music/executor/IExecutor.java
package me.wcy.music.executor;

/**
 * Created by hzwangchenyan on 2017/1/20.
 */
public interface IExecutor<T> {
    void execute();

    void onPrepare();

    void onExecuteSuccess(T t);

    void onExecuteFail(Exception e);
}

File: app/src/main/java/me/wcy/music/executor/IExecutor.kt
package me.wcy.music.executor

/**
 * Created by hzwangchenyan on 2017/1/20.
 */
interface IExecutor<T> {
    fun execute()
    fun onPrepare()
    fun onExecuteSuccess(t: T)
    fun onExecuteFail(e: Exception?)
}